.yamato/script/log_scripts/log_parser.py

```diff
@@ -0,0 +1,217 @@
+
+import argparse
+import os
+import sys
+import requests
+import json
+import glob
+import re
+from utils.execution_log_patterns import execution_log_patterns
+from utils.utr_log_patterns import utr_log_patterns
+
+'''
+This script runs for extended Yamato reporting. It
+1. Parses the execution log and for each command and its output (except successful non-retries and failed tests)
+    - finds matching patterns from execution_log_patterns
+    - sends to Yamato extended reporting server
+        title: command itself
+        summary: substring(s) of the command output matching the specified pattern(s)
+        tags: all (distinct) tags beloging to the matched pattern(s)
+        conclusion: failure/inconclusive/cancelled/success, which applies to the parsed command (not the whole job status)
+2. In case non-test related failure from UTR is matched, the script also reads TestResults.json and
+    - finds matching patterns from utr_log_patterns
+    - extends the data sent to Yamato extended reporting server in the previous point, by
+        extending the summary with matched substrings from the utr pattern and by
+        appending the utr pattern tags to the list of previous tags
+
+By default, the script requires no parameters and uses default execution log location in Yamato.
+To run it locally, specify
+    --local
+    --execution-log "<path to execution log file>"
+'''
+
+def load_json(file_path):
+    with open(file_path) as f:
+        json_data = f.readlines()
+        json_data[0] = json_data[0].replace('let testData = ','') # strip the beginning
+        json_data[-1] = json_data[-1][:-1] # strip the ; from end
+        return json.loads(' '.join(json_data))
+        #return json.load(f)
+
+def get_execution_log():
+    '''Returns the path to execution log file.'''
+    path_to_execution_log = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())),'Execution-*.log')
+    print('Searching for logs in: ', path_to_execution_log)
+
+    execution_log_file = glob.glob(path_to_execution_log)[0]
+    print('Reading log: ', execution_log_file)
+    return execution_log_file
+
+
+def read_execution_log(log_file_path):
+    '''Reads execution logs and returns:
+    logs: dictionary with keys corresponding to commands, and values containing log output and status
+    job_succeeded: boolean indicating if the job succeeded
+    '''
+
+    with open(log_file_path, encoding='utf-8') as f:
+        lines = [l.replace('\n','') for l in f.readlines() if l != '\n'] # remove empty lines and all newline indicators
+
+    # all log line idx starting/ending a new command
+    command_idxs = [i for i,line  in enumerate(lines) if '################################### Running next command ###################################' in line]
```

Comment:
Good point, yes it has been mentioned as a way of separation of commands by the Yamato team, but indeed we can't guarantee this will never change. However, I think if this happens, we will notice it quickly due to the script suddenly not working (it won't break the job though)